### PR TITLE
Upgrades `bump-my-version` and release commands from template

### PR DESCRIPTION
**Pull request by reedsa**
### What was wrong?

Outdated template

Related to Issue #
Closes #

### How was it fixed?

Merged latest from template with improved docs and release steps.

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-utils/blob/main/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -34,61 +41,73 @@ lint:
...
-check-bump:
-ifndef bump
-	$(error bump must be set, typically: major, minor, patch, or devnum)
-endif
+package-test: clean
+	python -m build
+	python scripts/release/test_package.py
 
-notes: check-bump validate-newsfragments
+notes: check-bump
```
Comment (kclowes) [inline Makefile:81]: running `make notes bump=patch` gives me a commit message of `v0.1.1`, so I don't think the eval is doing quite the right thing.

File: docs/index.rst
Comment (kclowes) [inline docs/index.rst:1]: This is much nicer!

File: pyproject.toml [hunk 2]
```diff
@@ -122,3 +123,46 @@ showcontent = true
 directory = "removal"
 name = "Removals"
 showcontent = true
+
+[tool.bumpversion]
+current_version = "0.1.0-alpha.0"
```
Comment (kclowes) [inline pyproject.toml:128]: I think this is why I got `v0.1.1` above. Should be 4.1.1

**Review by kclowes — APPROVED** [review]
Looks good! One version number got missed, I commented inline. If you're making changes to the Makefile, it's usually a good idea to manually run through each command you've changed to make sure they work as expected. The Makefile is a little finicky since at least the library name changes in each library, and sometimes it doesn't get caught until we're releasing. It's better to catch problems from the get-go.

File: pyproject.toml (hunk 2 above)
Comment (fselmo) [reply pyproject.toml:128]: +1, this needs to be changed to `4.1.1` before merge.

File: docs/index.rst
Comment (fselmo) [reply docs/index.rst:1]: agreed, this is real nice 👌🏼

**Review by fselmo — APPROVED** [review]
note: version change required in `pyproject.toml`

**Comment by reedsa** [comment]
Yep, that version fixed `make notes` and `make release` bump commands. Should be gtg!

**Review by kclowes — APPROVED** [review]
🚀

**Pull request merged**